src/lib/dispatcher.js

```diff
@@ -42,7 +42,7 @@ async function archiveDirectory(basecommand, sourceDir, outputFilePath) {
 
     output.on('close', function () {
       const zipSizeBytes = archive.pointer();
-      basecommand.log(
+      basecommand.doLog(
```

Comment:
I think we could get away with passing a logging function instead of `basecommand`. But that's an unrelated improvement.